sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/mapper/ChangeStreamRecordMapper.java

```diff
@@ -260,6 +312,65 @@ private DataChangeRecord toDataChangeRecord(
         changeStreamRecordMetadataFrom(partition, commitTimestamp, resultSetMetadata));
   }
 
+  private DataChangeRecord toDataChangeRecordJson(
+      PartitionMetadata partition, Value row, ChangeStreamResultSetMetadata resultSetMetadata) {
+    Value dataChangeRecordValue =
+        Optional.ofNullable(row.getStructValue().getFieldsMap().get(DATA_CHANGE_RECORD_COLUMN))
+            .orElseThrow(IllegalArgumentException::new);
+    Map<String, Value> valueMap = dataChangeRecordValue.getStructValue().getFieldsMap();
+    final String commitTimestamp =
+        Optional.ofNullable(valueMap.get(COMMIT_TIMESTAMP_COLUMN))
+            .orElseThrow(IllegalArgumentException::new)
+            .getStringValue();
+    return new DataChangeRecord(
+        partition.getPartitionToken(),
+        Timestamp.parseTimestamp(commitTimestamp),
+        Optional.ofNullable(valueMap.get(SERVER_TRANSACTION_ID_COLUMN))
+            .orElseThrow(IllegalArgumentException::new)
```

Comment:
All the fields that we are doing `orElseThrow` are mandatory right (just double checking)?

Reply:
Yes, I doublechecked.

See JSON string for data change record:     Jsonb: {"data_change_record":{"column_types":[{"is_primary_key":true,"name":"SingerId","ordinal_position":1,"type":{"code":"INT64"}},{"is_primary_key":false,"name":"FirstName","ordinal_position":2,"type":{"code":"STRING"}},{"is_primary_key":false,"name":"LastName","ordinal_position":3,"type":{"code":"STRING"}},{"is_primary_key":false,"name":"SingerInfo","ordinal_position":4,"type":{"code":"BYTES"}}],"commit_timestamp":"2022-11-29T11:31:36.008927-08:00","is_last_record_in_transaction_in_partition":true,"is_system_transaction":false,"mod_type":"INSERT","mods":[{"keys":{"SingerId":"2"},"new_values":{"FirstName":"First Name 2","LastName":"Last Name 2","SingerInfo":null},"old_values":{}}],"number_of_partitions_in_transaction":1,"number_of_records_in_transaction":1,"record_sequence":"00000000","server_transaction_id":"NTczMzU4MTQ3MzQyNzAyNjA0Ng==","table_name":"Singers_qqkl32uoQ1Rhk4ZDqW4KGy2f1oGyhLH6oiiXq71B5U24ZLbTylFANYz","transaction_tag":"","value_capture_type":"OLD_AND_NEW_VALUES"}}


See JSON string for child partitions record:     Jsonb: {"child_partitions_record":{"child_partitions":[{"parent_partition_tokens":[],"token":"__8BAYEGX9zUgAABgsBFg1NpbmdlcnNTdHJlYW1fODVxa29vZHBkM3ZOQUFvOAABhIEGCRQ28AANgoCDCMNkAAAAAAAAhASSRgwFhWcxM185NTE5OTgzAAH__4X_Be6gldXMFYb_Be6hGsHih4eAwGQBAf__"}],"record_sequence":"00000002","start_timestamp":"2022-11-29T11:31:35.951359-08:00"}}

Reply:
Thanks